src/components/data-list-old/index.tsx

```diff
@@ -0,0 +1,322 @@
+//Old implementation - this file will be updated
+import { useEffect, useState } from "react";
+import { Button } from "@mui/material";
+import { Download, Upload } from "@mui/icons-material";
+import {
+  ModuleContainer,
+  ModuleHeaderContainer,
+  ModuleHeaderSubtitleContainer,
+} from "components/module";
+import { CoreModule, getAddFormRoute } from "lib/router";
+import { GhostLink } from "components/ghost-link";
+import {
+  ActionsContainer,
+  AddButtonContainer,
+  ExtraActionsContainer,
+  LeftContainer,
+  RightContainer,
+} from "./index.styled";
+import {
+  defaultFilterLimit,
+  getBasicExportFilterQuery,
+  getBasicFilterQuery,
+  getWhereFilterQuery,
+  totalCountHeaderName,
+} from "lib/query";
+import { BreadCrumbNavigation } from "components/breadcrumbs";
+import { CsvImport } from "components/spreadsheet-import";
+import { Result } from "@wavepoint/react-spreadsheet-import/types/types";
+import { CsvExport } from "components/export";
+import { SearchBar } from "components/search-bar";
+import { DataTableGrid } from "components/data-table";
+import { GridColDef, GridColumnVisibilityModel, GridSortDirection } from "@mui/x-data-grid";
+import { GridInitialStateCommunity } from "@mui/x-data-grid/models/gridStateCommunity";
+import { getModelByName } from "lib/network/swagger-models";
+import { BreadcrumbLink } from "utils/types";
+
+type dataListProps = {
+  modelName: string;
+  columns: GridColDef<any>[];
+  dataListBreadcrumbLinks: BreadcrumbLink[];
+  currentBreadcrumb: string;
+  searchBarLabel: string;
+  defaultFilterOrderColumn: string;
+  defaultFilterOrderDirection: string;
+  initialGridState: GridInitialStateCommunity | undefined;
+  exportFileName: string;
+  endRoute: string;
+  getModelDataList: (query: string) => any;
+  exportAsync: (query: string, accept: string) => Promise<string>;
+  dataImportCreate: (data: any) => void;
+};
+
+type dataListSettings = {
+  searchTerm: string;
+  filterLimit: number;
+  skipLimit: number;
+  sortColumn: string;
+  sortOrder: string;
+  whereField: string;
+  whereFieldValue: string;
+  pageNumber: number;
+  columnVisibilityModel: GridColumnVisibilityModel;
+};
+
+export const DataList = ({
+  modelName,
+  columns,
+  dataListBreadcrumbLinks,
+  currentBreadcrumb,
+  searchBarLabel,
+  defaultFilterOrderColumn,
+  defaultFilterOrderDirection,
+  initialGridState,
+  endRoute,
+  getModelDataList,
+  exportAsync: exportAsync,
+  exportFileName,
+  dataImportCreate,
+}: dataListProps) => {
+  const [modelData, setModelData] = useState<any[] | undefined>([]);
```

Comment:
Maybe it's worth to consolidate multiple states into single object as far as i can understand every change of any of these variables causes whole object re-render so uniting them wouldn't hurt performance but will save some memory and will decrease CPU usage

Reply:
Yeah. Changes are done now but it will be merged with next PR due to some other dependencies in common components.